src/vaadin-menu-bar-buttons-mixin.html

```diff
@@ -18,19 +23,48 @@
   /**
    * @polymerMixin
    */
-  Vaadin.MenuBar.ButtonsMixin = superClass => class MenuButtonsMixin extends superClass {
+  Vaadin.MenuBar.ButtonsMixin = superClass => class extends Polymer.mixinBehaviors(Polymer.IronResizableBehavior, superClass) {
+
+    static get properties() {
+      return {
+        _hasOverflow: {
+          type: Boolean,
+          value: false
+        }
+      };
+    }
+
+    static get observers() {
+      return [
+        '_buttonsRendered(items, items.splices)'
+      ];
+    }
+
+    constructor() {
+      super();
+      this._hiddenButtons = new WeakMap();
```

Comment:
Why isn't this declared/initiated normally in the properties object`.

What's the biggest benefit in using a Map instead of an array of widths? Currently, the width is fetched in various places by using an index to get a reference to a button (from `_buttons`) and then the width is obtained from the map by that button.

Reply:
> Why isn't this declared/initiated normally in the properties object
This property isn't intended to be used anywhere in the template / observers / computed properties etc, so it doesn't need any accessors to be defined by Polymer.

> What's the biggest benefit in using a Map instead of an array of widths? 

One possible edge case which came to my mind is changing the "items" array while certain root menu buttons are hidden. 

I'm not completely sure if WeakMap is better than array, though. Since `dom-repeat` is reusing nodes, and doesn't have a `restamp` option, the references in WeakMap won't be GCed anyways. So I can try to rewrite the same code using arrays.

Reply:
Yea, as you mentioned, since dom-repeat reuses the nodes, we might end up having erroneous width values in the map when there are changes to `items`. Could be that there's no way to skip updating the width values on changes to `items.*`.

One approach might be to only set the overflowing buttons `visibility: hidden; position: absolute;` (instead of `display: none`) so they could be measured dynamically.

Reply:
Updated as you suggested, also had to make non-visible buttons disabled.